lambda/common/src/main/scala/com/gu/hq/AWS.scala

```diff
@@ -27,31 +28,37 @@ object AWS {
     new ProfileCredentialsProvider("developerPlayground")
   )
 
-  // EC2
-  def ec2client(region: Regions): AmazonEC2 = {
-    AmazonEC2AsyncClientBuilder.standard().withRegion(region).withCredentials(securityCredentialsProviderChain).build()
-  }
-
   // ELBs
-  def elbClient(region: Regions) = {
+  def elbClient(region: Regions): AmazonElasticLoadBalancingAsync = {
     AmazonElasticLoadBalancingAsyncClient.asyncBuilder().withRegion(region).withCredentials(securityCredentialsProviderChain).build()
   }
 
   // SNSs
-  def snsClient(region: Regions) = {
-    AmazonSNSAsyncClient.asyncBuilder().withRegion(region).withCredentials(developerPlaygroundCredentialsProviderChain ).build()
+  def snsClient(region: Regions): AmazonSNSAsync = {
+    AmazonSNSAsyncClient.asyncBuilder().withRegion(region).withCredentials(securityCredentialsProviderChain ).build()
   }
   // STS
-  def stsClient(region: Regions) = {
+  def stsClient(region: Regions): AWSSecurityTokenServiceAsync = {
     AWSSecurityTokenServiceAsyncClient.asyncBuilder().withRegion(region).withCredentials(securityCredentialsProviderChain ).build()
   }
 
+  // S3
+  def s3Client(region: Regions): AmazonS3 = {
+    AmazonS3ClientBuilder.standard().withRegion(region).withCredentials(securityCredentialsProviderChain).build()
+  }
+
   def describeLoadBalancers(elbClient: AmazonElasticLoadBalancingAsync): DescribeLoadBalancersResult = {
     val request = new DescribeLoadBalancersRequest()
     elbClient.describeLoadBalancers(request)
   }
 
-  def accountNumber(stsClient: AWSSecurityTokenServiceAsync) = {
+  def accountNumber(stsClient: AWSSecurityTokenServiceAsync): String = {
     stsClient.getCallerIdentity(new GetCallerIdentityRequest()).getAccount
   }
+
+  def accountName(s3Client: AmazonS3, accountNumber: String):String = {
+    def accountsJson = scala.io.Source.fromInputStream(s3Client.getObject("guardian-dist", "guardian/PROD/accounts").getObjectContent).mkString
```

Comment:
We're mixing logic and IO here. It'd be great if we could test this code?

Fetching the AWS account name config should be a separate step to the logic of getting the account name. This function should *just* do the IO. All the rest can be pure tested code.

Reply:
Done.